pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1533,40 +1591,49 @@ func (gce *GCECloud) DeleteRoute(clusterName string, route *cloudprovider.Route)
 }
 
 func (gce *GCECloud) GetZone() (cloudprovider.Zone, error) {
-	region, err := getGceRegion(gce.zone)
-	if err != nil {
-		return cloudprovider.Zone{}, err
-	}
 	return cloudprovider.Zone{
-		FailureDomain: gce.zone,
-		Region:        region,
+		FailureDomain: gce.instanceZone,
+		Region:        gce.region,
 	}, nil
 }
 
-func (gce *GCECloud) CreateDisk(name string, sizeGb int64) error {
+// Create a new Persistent Disk, with the specified name & size.
+// If zone is specified (!= ""), the disk will be created in that zone.
+// If the zone is not specified (""), the disk will be created in the zone in which we are currently running.
+// Currently, this will be the zone where the kube-controller-manager is running.
+// TODO: We should support auto-provisioning volumes in multiple/specified zones
+func (gce *GCECloud) CreateDisk(name string, zone string, sizeGb int64) error {
 	diskToCreate := &compute.Disk{
 		Name:   name,
 		SizeGb: sizeGb,
 	}
-	createOp, err := gce.service.Disks.Insert(gce.projectID, gce.zone, diskToCreate).Do()
+	if zone == "" {
```

Comment:
I really don't like an empty string here.  Ideally get the caller to pass in the sone name, or perhaps nil, if that's not feasible.